benchmark/single-source/SetTests.swift

```diff
@@ -12,107 +12,156 @@
 
 import TestsUtils
 
-public let SetTests = [
-  BenchmarkInfo(name: "SetExclusiveOr", runFunction: run_SetExclusiveOr, tags: [.validation, .api, .Set]),
-  BenchmarkInfo(name: "SetExclusiveOr_OfObjects", runFunction: run_SetExclusiveOr_OfObjects, tags: [.validation, .api, .Set]),
-  BenchmarkInfo(name: "SetIntersect", runFunction: run_SetIntersect, tags: [.validation, .api, .Set]),
-  BenchmarkInfo(name: "SetIntersect_OfObjects", runFunction: run_SetIntersect_OfObjects, tags: [.validation, .api, .Set]),
-  BenchmarkInfo(name: "SetIsSubsetOf", runFunction: run_SetIsSubsetOf, tags: [.validation, .api, .Set]),
-  BenchmarkInfo(name: "SetIsSubsetOf_OfObjects", runFunction: run_SetIsSubsetOf_OfObjects, tags: [.validation, .api, .Set]),
-  BenchmarkInfo(name: "SetUnion", runFunction: run_SetUnion, tags: [.validation, .api, .Set]),
-  BenchmarkInfo(name: "SetUnion_OfObjects", runFunction: run_SetUnion_OfObjects, tags: [.validation, .api, .Set]),
-]
-
-@inline(never)
-public func run_SetIsSubsetOf(_ N: Int) {
-  let size = 200
+let size = 400
+let overlap = 100
 
-  SRand()
+let setAB = Set(0 ..< size)
+let setCD = Set(size ..< 2 * size)
+let setBC = Set(size - overlap ..< 2 * size - overlap)
+let setB = Set(size - overlap ..< size)
 
-  var set = Set<Int>(minimumCapacity: size)
-  var otherSet = Set<Int>(minimumCapacity: size)
+let setOAB = Set(setAB.map(Box.init))
+let setOCD = Set(setCD.map(Box.init))
+let setOBC = Set(setBC.map(Box.init))
+let setOB = Set(setB.map(Box.init))
 
-  for _ in 0 ..< size {
-    set.insert(Int(truncatingIfNeeded: Random()))
-    otherSet.insert(Int(truncatingIfNeeded: Random()))
-  }
-
-  var isSubset = false
-  for _ in 0 ..< N * 5000 {
-    isSubset = set.isSubset(of: otherSet)
-    if isSubset {
-      break
-    }
-  }
+let countAC = 2 * (size - overlap)
+let countABC = 2 * size - overlap
+let countABCD = 2 * size
+let countB = overlap
 
-  CheckResults(!isSubset)
-}
+public let SetTests = [
+  BenchmarkInfo(
+    name: "SetExclusiveOr",
+    runFunction: { n in run_SetExclusiveOr(setAB, setCD, countABCD, 100 * n) },
+    tags: [.validation, .api, .Set],
+    setUpFunction: { blackHole([setAB, setCD]) }),
+  BenchmarkInfo(
+    name: "SetExclusiveOr2",
+    runFunction: { n in run_SetExclusiveOr(setAB, setBC, countAC, 10 * n) },
+    tags: [.validation, .api, .Set],
+    setUpFunction: { blackHole([setAB, setBC]) }),
+  BenchmarkInfo(
+    name: "SetExclusiveOr_OfObjects",
```

Comment:
Could you please drop the underscores from test names, at least from the new ones? I'd call them like `SetExclusiveOrBoxed`, if coming up with the name from scratch.

Reply:
Sure, as long as we end up losing compatibility. But if some of the original names remain, I'd prefer to make the relationship between old/new tests obvious. (Naming consistency is far less important than utility in this case.)

Reply:
@eeckstein cares a lot about continuity of perf-tracking… I might have a look at adding some `lnt` migration scripts for that if it remains a point of contention.